File: YoutubeRadioApplication/src/main/java/com/ldz/project/service/music/inter/IMusicService.java
package com.ldz.project.service.music.inter;

import com.ldz.music.manager.model.UserMusicStatusDTO;

/**
 * Created by ldalzotto on 03/05/2017.
 */
public interface IMusicService {

    public Boolean addMusicFromUrlAndMusicPlatform(String username, String url, String musicPlatform);

    public UserMusicStatusDTO getMusicStatusOfCurrentClient(String username);
}

File: YoutubeRadioApplication/src/scala/com/ldz/project/service/music/IMusicService.scala
package com.ldz.project.service.music

import com.ldz.music.manager.model.UserMusicStatusDTO

/**
  * Created by ldalzotto on 23/06/2017.
  */
trait IMusicService {
  def addMusicFromUrlAndMusicPlatform(username: String, url: String, musicPlatform: String): Boolean

  def getMusicStatusOfCurrentClient(username: String): UserMusicStatusDTO

  def getAllDistinctArtistsFromClient(username: String): Seq[String]

  def getFilteredArtistsFromInput(username: String, musicInput: String): Seq[String]
}
